pkg/sql/schemachanger/scplan/internal/rules/dep_drop_object.go

```diff
@@ -26,8 +26,36 @@ import (
 //   reaches the ABSENT state, but afterwards in the stage, so as to not
 //   interfere with the event logging op which is tied to the descriptor element
 //   removal.
+
 func init() {
 
+	registerDepRule(
+		"descriptor txn_drop before descriptor drop",
+		scgraph.PreviousStagePrecedence,
+		"descriptor", "dependent",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.typeFilter(IsDescriptor),
+				from.el.AttrEqVar(screl.DescID, "_"),
+				from.el.AttrEqVar(rel.Self, to.el),
+				to.el.AttrEqVar(rel.Self, to.el),
```

Comment:
Isn't this line always true and hence redundant?

Reply:
Yes, removed